Update mobile nav to be on the side

enejb opened the pull request:
#### Changes proposed in this Pull Request:

This PR update the mobile nav on the following themes. The update is to make the theme more streamlines by moving the menu to the right side of the site title and description.

For example: 
Before:
<img width="492" alt="Screen Shot 2020-11-30 at 2 26 28 PM" src="https://user-images.githubusercontent.com/115071/101228220-94b2e380-364f-11eb-8e0d-b5d9d2a512c8.png">

After:
<img width="483" alt="Screen Shot 2020-11-30 at 2 28 32 PM" src="https://user-images.githubusercontent.com/115071/101228218-9381b680-364f-11eb-8509-90815c595c0b.png">


Since the themes are already available on site on WordPress.com and we don't want to change the navigation on those sites. We will have to overwrite the `function varia_mobile_nav_on_side( $classes )` in a mu-plugin by defining that function before the one that is provided by the theme. 


**Alves**
<details><summary>Screenshots</summary>
<img width="528" alt="Screen Shot 2020-12-04 at 2 20 14 PM" src="https://user-images.githubusercontent.com/115071/101225714-05a1cd80-3647-11eb-8020-dce423bcdaae.png">
<img width="528" alt="Screen Shot 2020-12-04 at 2 20 09 PM" src="https://user-images.githubusercontent.com/115071/101225717-06d2fa80-3647-11eb-8fe5-1c3b6a4589d5.png">

</details>

**Barnsbury**
<details><summary>Screenshots</summary>
<img width="533" alt="Screen Shot 2020-12-04 at 2 19 24 PM" src="https://user-images.githubusercontent.com/115071/101225826-6204ed00-3647-11eb-94fb-a98355b5826f.png">
<img width="548" alt="Screen Shot 2020-12-04 at 2 19 14 PM" src="https://user-images.githubusercontent.com/115071/101225829-629d8380-3647-11eb-91e5-de8d75abfe32.png">
</details>

**Dalson**
<details><summary>Screenshots</summary>
<img width="506" alt="Screen Shot 2020-12-04 at 3 13 40 PM" src="https://user-images.githubusercontent.com/115071/101225871-7e088e80-3647-11eb-94dd-c0bab8273420.png">
<img width="504" alt="Screen Shot 2020-12-04 at 3 13 35 PM" src="https://user-images.githubusercontent.com/115071/101225873-7e088e80-3647-11eb-82d6-edee967c0cfd.png">

</details>

**Hever**
<details><summary>Screenshots</summary>
<img width="526" alt="Screen Shot 2020-12-04 at 2 54 22 PM" src="https://user-images.githubusercontent.com/115071/101228175-5e756400-364f-11eb-8e57-06580f932471.png">
<img width="530" alt="Screen Shot 2020-12-04 at 2 54 11 PM" src="https://user-images.githubusercontent.com/115071/101228176-5e756400-364f-11eb-98ed-bde0f244d77d.png">

</details>

**Mayland**
<details><summary>Screenshots</summary>
<img width="523" alt="Screen Shot 2020-12-04 at 2 43 27 PM" src="https://user-images.githubusercontent.com/115071/101228161-4ac9fd80-364f-11eb-831f-689f03fce794.png">
<img width="533" alt="Screen Shot 2020-12-04 at 2 43 20 PM" src="https://user-images.githubusercontent.com/115071/101228162-4b629400-364f-11eb-903f-487b006ddb14.png">

</details>

**Morden**
<details><summary>Screenshots</summary>
<img width="507" alt="Screen Shot 2020-12-04 at 3 24 10 PM" src="https://user-images.githubusercontent.com/115071/101228150-400f6880-364f-11eb-9373-36f604f316d6.png">
<img width="516" alt="Screen Shot 2020-12-04 at 3 24 05 PM" src="https://user-images.githubusercontent.com/115071/101228151-40a7ff00-364f-11eb-83d6-cba34364eda9.png">

</details>

**Rivington**
<details><summary>Screenshots</summary>
<img width="521" alt="Screen Shot 2020-12-04 at 2 49 30 PM" src="https://user-images.githubusercontent.com/115071/101225896-8d87d780-3647-11eb-8c8d-6837f321d62e.png">
<img width="521" alt="Screen Shot 2020-12-04 at 2 49 24 PM" src="https://user-images.githubusercontent.com/115071/101225897-8e206e00-3647-11eb-91b4-6a88c40cb0fc.png">

</details>


#### Related issue(s):
See pNEWy-doZ-p2

enejb committed:
Update the Alves theme to have the mobile nav on the side.

enejb committed:
Update the Barnsbury theme to have the mobile nav on the side.

enejb committed:
Update the Mayland theme to have the mobile nav on the side.

enejb committed:
Update the Rivington theme to have the mobile nav on the side.

enejb committed:
Update the Hever theme to have the mobile nav on the side.

enejb committed:
Update the Dalston theme to have the mobile nav on the side.

enejb committed:
Update the Morden theme to have the mobile nav on the side.

enejb committed:
Include the varia_mobile_nav_on_side function in all the themes.

This makes sure that the user will not get a PHP warning if theu are not running the latest version of varia that has that function.
Since varia doesn't have that feature enabled this PR removes the function as well.

kathrynwp commented:
It would be great if you could please make sure this also looks OK when the site has a logo set. Thanks!

kathrynwp commented:
This change could also affect users with custom CSS, so would be good to give Happiness and the CSS guild a heads-up once deployed.

enejb commented:
Thanks for raising the issues regarding site logos and custom css @kathrynwp. 

I haven' tested the change with a custom logo yet. I will do so today.

The plan for the roll out is to not change sites that have the theme applied already on .com but to only apply the change to new sites. I am not sure what we can can do for .org sites or atomic sites. 🤔 So the issue with custom css shouldn't be a problem since sites should maintain the same css as before.

scruffian commented:
The CSS looks the same for every child theme - is that correct? If so then i'd add the CSS to Varia even though its not used in that theme, and that way we reduce the duplication - each child will inherit the CSS anyway...

scruffian commented on alves/functions.php:
```diff
@@ -231,3 +233,10 @@ function alves_editor_styles() {
 add_action( 'enqueue_block_editor_assets', 'alves_editor_styles' );
+
+// This makes it possible to define the function in earlier to alter if the class should be applied or not.
+if ( ! function_exists( 'varia_mobile_nav_on_side' ) ) {
```
I think we can also add this to Varia, then the child theme only needs to add the one liner for the filter.

enejb replied:
I was thinking about this as well. Initially I had the function defined in `varia/functions.php` the issue that I ran into is that if the site doesn't update the parent theme (varia) to the latest version. We end up with the site creating PHP warning which is no a issue on .com but maybe atomic and .org sites. Since as far as I know don't have a good way to also update the parent theme when you update the child theme. 

So I decided to move the functions to the theme's functions.php file.

scruffian replied:
Varia themes aren't on .org. In theory it should be fine on Atomic too, as the parent theme should get updates when the child does. You might want to double check that with the Atomic team though.

scruffian replied:
Also we can just wrap the filter in a check for the function and avoid the warning that way.

enejb committed:
Updates to take into accout the site logo & when the menu is not there.

enejb committed:
Updates to take into accout the theme switch.

enejb committed:
Update scss files to be more DRY

enejb commented:
> The CSS looks the same for every child theme - is that correct? If so then i'd add the CSS to Varia even though its not used in that theme, and that way we reduce the duplication - each child will inherit the CSS anyway...

@scruffian I updated the sass files to that they are the code that repeats itself is just in once place.

scruffian commented on barnsbury/sass/_extra-child-theme.scss:
```diff
@@ -393,3 +393,6 @@ a {
 }
+// Updates the Mobile Navigation to be next to the site title.
+$mobile-nav-side-right-margin: 100px;
+@import "../../varia/sass/components/header/_site-mobile-nav-side";
```
This approach also relies on the parent theme being updated.

Do we need to import the CSS here? Can't we just add the CSS itself to the parent theme and then add one rule to change the right margin for each child?

enejb replied:
The imported css here gets expanded and and combined into the css that ships with the child themes. 
So it does rely on the parent theme but only during the development. So I think it is fine in this case.

scruffian replied:
Oh good point :)

scruffian commented:
I think it would be worth making this an option in the Customizer, so that users who have it set up the old way can change it. 

You can see an example of how to do that here: https://github.com/Automattic/themes/blob/0100e61255bc8175d354426db440006b75e4eb2c/spearhead/inc/wpcom-colors.php#L30

enejb committed:
Move functions to Varia add customizer control

enejb committed:
Move functions to after theme setup call

MaggieCabrera commented:
I tried this and it worked fine for me, but I wonder if we want to tweak this a bit when a logo is active on the site (this is Barnsbury Theme): 

<img width="425" alt="Screenshot 2020-12-09 at 11 39 47" src="https://user-images.githubusercontent.com/3593343/101619314-66e4ea80-3a13-11eb-9da7-78ff487b9861.png">

Edit: tagging @kjellr for design input

MaggieCabrera commented on barnsbury/style-rtl.css:
```diff
@@ -4390,3 +4390,32 @@ body.admin-bar .widget_eu_cookie_law_widget.widget.top {
+	.mobile-nav-side .has-menu .site-logo,
+	.mobile-nav-side .has-menu .site-title,
+	.mobile-nav-side .has-menu .site-description {
+		margin-left: 100px;
```
I'm worried if this value will work if the button changes sizes. I'm not sure if other languages will change the width of the button.

scruffian commented on varia/sass/components/header/_site-mobile-nav-side.scss:
```diff
@@ -0,0 +1,31 @@
+// Updates the Mobile Navigation to be next to the site title.
```
nitpick: Looks like this file has spaces not tabs

scruffian committed:
Move the option to the menu setion

scruffian commented:
I added a commit to move the option to the menus section as that made more sense in retrospect.

scruffian commented on alves/functions.php:
```diff
@@ -102,6 +102,21 @@ function alves_setup() {
+
+		// Enable the mobile nav on side on theme switch.
+		if ( function_exists( 'varia_enable_mobile_nav_on_side' ) ) {
+			add_action( 'after_switch_theme', 'varia_enable_mobile_nav_on_side' );
```
This is great

scruffian commented:
This is looking good. I think @MaggieCabrera  is right though we can't use absolute positioning for the menu button as we can't be sure how wide it will be. I'd use flex box. Let us know if you need any help with it. Thanks for all your work on this.

kjellr commented:
> I wonder if we want to tweak this a bit when a logo is active on the site (this is Barnsbury Theme):

> This is looking good. I think @MaggieCabrera is right though we can't use absolute positioning for the menu button as we can't be sure how wide it will be. I'd use flex box. Let us know if you need any help with it. Thanks for all your work on this.

Yeah, if we can use flexbox (or grid) there, it should take care of the problem.

scruffian committed:
Use CSS grid to align the menu

enejb committed:
Revert "Use CSS grid to align the menu"

This reverts commit 8ab0d49cc7e4b6e7b77b5df7f1bc5fb494c7166c.

enejb committed:
Use css grid to style the side nav

enejb committed:
After rebuild the css again.

enejb committed:
Possible woo plugin css.

scruffian commented on varia/sass/components/header/_site-mobile-nav-side.scss:
```diff
@@ -0,0 +1,86 @@
+				margin-top: -2px;
+			}
+
+			.menu-main-menu-container {
```
This class is different depending on the name of the menu. I think we'll need to target it with `> div`

enejb replied:
This is now fixed

enejb committed:
Fix the menu container class.

enejb committed:
Only add the container class if we are in the new enable_side_menu mode since it could casue issues with backwards compatiblity and css

MaggieCabrera committed:
used template areas for menu layout

MaggieCabrera committed:
styles for dropdown menu

MaggieCabrera commented:
I have been looking into this issue and I went ahead and pushed a few changes to the original approach. After talking with @scruffian it seemed simpler to go with `grid-template-areas` and the solution lends itself to adaptation for the themes that have a slightly different html.

I have found some problems with this approach that are not easily solvable without changing the html. The first one is that the menu needs to use position absolute and moved out of the flow. This causes the open menu to cover the social menu and sometimes the site tagline if the font size is big (it happens in Alves). I tried to maintain the general design of the dropdown menu but I believe we need some design input before we go any further. @kjellr if you can have a look at this, it would be very helpful, thank you.

Lastly, since we are applying a background color to the menu that wasn't needed before, we need to take it into account for color variations. I have left that out of my commits and will get to it after we are done with any design changes in case that interferes in any way. I'm pointing this out here as a reminder to do that before we are done with this PR.

### Screenshots

There are several possibilities that come into play here:

Only logo | Only title + tagline
--- | ---
<img width="433" alt="Screenshot 2020-12-11 at 12 20 56" src="https://user-images.githubusercontent.com/3593343/102069997-d1749c80-3dfe-11eb-975e-46e7d14a850f.png"> | <img width="440" alt="Screenshot 2020-12-11 at 12 20 39" src="https://user-images.githubusercontent.com/3593343/102069949-bc980900-3dfe-11eb-811e-80ca0a51850b.png">

No logo, no title | Logo + Title and tagline
--- | ---
<img width="427" alt="Screenshot 2020-12-11 at 12 21 41" src="https://user-images.githubusercontent.com/3593343/102070037-dd605e80-3dfe-11eb-92e5-7ec8f98e63a3.png"> | <img width="486" alt="Screenshot 2020-12-14 at 11 25 49" src="https://user-images.githubusercontent.com/3593343/102070255-26b0ae00-3dff-11eb-8e0a-9a92128fcbb9.png">


Here we can see the problem with the overlap of the menu in **Alves**:


<img width="410" alt="Screenshot 2020-12-14 at 11 15 42" src="https://user-images.githubusercontent.com/3593343/102070304-3af4ab00-3dff-11eb-964c-257c0e8d2bf3.png">

scruffian commented on varia/sass/components/header/_site-mobile-nav-side.scss:
```diff
@@ -0,0 +1,107 @@
+				margin-top: -2px;
+			}
+
+			.main-menu-container {
```
I don't think we can use class names here as they change for each menu. The way we do it in other themes is with a `>div` selector.

MaggieCabrera replied:
Changed it

MaggieCabrera committed:
changed menu container selector

kjellr commented:
> I have found some problems with this approach that are not easily solvable without changing the html. The first one is that the menu needs to use position absolute and moved out of the flow. This causes the open menu to cover the social menu and sometimes the site tagline if the font size is big (it happens in Alves). I tried to maintain the general design of the dropdown menu but I believe we need some design input before we go any further. @kjellr if you can have a look at this, it would be very helpful, thank you.

Thanks for digging into this, @MaggieCabrera. Outside of changing the HTML (which I understand we don't want to do), I think the next-best solution would be to have the expanded menu take up the full viewport. This gets a little weird when a header logo are involved, but it still works: 

No Site Logo|With Site Logo
---|---
![Kapture 2020-12-14 at 13 28 41](https://user-images.githubusercontent.com/1202812/102121160-81eb9c00-3e11-11eb-912e-f01af3fc9c04.gif)|![Kapture 2020-12-14 at 13 35 31](https://user-images.githubusercontent.com/1202812/102121186-90d24e80-3e11-11eb-9dfa-6ee3c3c14c25.gif)

We would need to prevent the background from scrolling when this menu is open, but I _think_ that should be doable.

MaggieCabrera commented:
> We would need to prevent the background from scrolling when this menu is open, but I think that should be doable.

I think we can do that, thanks for the input! In the cases where the woocommerce cart button is present, I assume we want to hide it, right? In fact we should swap the order from left to right the way the two buttons are now, so that the cart one is to the left of the menu button, does that make sense?

kjellr commented:
Yeah, the WooCommerce button should be hidden when the menu is open. And I agree that it makes sense to show the Menu button to the right of the Woo Cart. 👍

enejb commented:
Thanks for all the changes @MaggieCabrera ! 

It is starting to look really nice. 

I found that 2 themes. Rivinstong and Mayland didn't quite follow the same layout. 

See 
| Mayland   | Rivinston      |  
|----------|:-------------:|
|  <img width="459" alt="Screen Shot 2020-12-14 at 3 28 18 PM" src="https://user-images.githubusercontent.com/115071/102148613-a0599380-3e21-11eb-9de7-841ce0ceba0b.png"> | <img width="405" alt="Screen Shot 2020-12-14 at 3 35 37 PM" src="https://user-images.githubusercontent.com/115071/102148918-35f52300-3e22-11eb-91a4-bc69faeedaac.png"> |

Notice that the site title and the site description take up the 100% of the column. 

On the  Barnsbury, Daltson, Hever and Morden themes the Woo menus ends below the current Menu button. For example. 

<img width="512" alt="Screen Shot 2020-12-14 at 3 28 56 PM" src="https://user-images.githubusercontent.com/115071/102149025-6ccb3900-3e22-11eb-81cd-732fc4f2016b.png"> Morden theme. 

Would it be possible but can we try to align them to be on the same line. I think this might not be possible with all the different language combination out there though.

Thanks again for your help with this!

MaggieCabrera committed:
changed mobile menu to be fixed and cover all the screen

MaggieCabrera committed:
lock scroll when mobile menu is open

MaggieCabrera committed:
considered scrolling menu, changed order of buttons

MaggieCabrera committed:
avoid wrap on navigation

MaggieCabrera committed:
compiled styles

MaggieCabrera commented:
Working on the issue pointed by @enejb for Rivington and Mayland I'm a bit blocked by the wrapper div (.site-branding) to make it work with the grid approach. I have found out a very nifty rule: `display: contents;` This rule lets the children of the element act as if they were direct children of their "grandparent". The drawback: [it doesn't work in IE](https://caniuse.com/css-display-contents). My latest commits include this solution for testing.


Before | After 
--- | ---
<img width="435" alt="Screenshot 2020-12-16 at 10 59 05" src="https://user-images.githubusercontent.com/3593343/102334236-b9cc1e00-3f8e-11eb-9378-269663eb6c12.png"> | <img width="469" alt="Screenshot 2020-12-16 at 10 58 58" src="https://user-images.githubusercontent.com/3593343/102334245-bb95e180-3f8e-11eb-9832-8f2e28cd605d.png">


The options I see to go around this are:

- [ ] Keep the `display: contents;` solution and don't support IE. It doesn't look broken, we can keep the "before" option for IE and the "after" for the rest of the browsers. 
- [ ] Solve this with JS. I don't like this idea.
- [ ] Remove the wrapper and adapt the CSS on desktop so the styles aren't affected by that. This is potentially complicated.

MaggieCabrera committed:
display contents solution

scruffian commented:
This looks great. I noticed is that the menu button moves slightly when it's opened, like this:
![variamenu](https://user-images.githubusercontent.com/275961/102344799-59d87600-3f94-11eb-878d-0520ce0734fd.gif)

kjellr commented:
Thanks Maggie! I tested in Alves, and noticed a couple bugs: 

![Kapture 2020-12-16 at 09 03 16](https://user-images.githubusercontent.com/1202812/102358529-9ba9f200-3f7d-11eb-8111-8ebd5fff2fbf.gif)

- The first is the same thing that @scruffian noted — the menu button jumps a little but when it's open. 
- And the second is that the cart menu appears to be clipped and scrolling in a tiny area. 

---

On Rivington, I noticed that the cart button appears taller than the menu button for some reason, but only if I have a site logo: 

![Screen Shot 2020-12-16 at 9 07 02 AM](https://user-images.githubusercontent.com/1202812/102358962-2e4a9100-3f7e-11eb-99be-d490420c18a4.png)

---

> Keep the display: contents; solution and don't support IE. It doesn't look broken, we can keep the "before" option for IE and the "after" for the rest of the browsers.

Wow, `display: contents` is weird! If we can fallback to the old way of doing this for IE (left-aligning the menu), then I think we're ok to keep using this rule here. It does seem to solve things in my testing.

MaggieCabrera commented:
> Wow, display: contents is weird! If we can fallback to the old way of doing this for IE (left-aligning the menu), then I think we're ok to keep using this rule here. It does seem to solve things in my testing.

I had an aha moment when I noticed that this is a non-issue, since this rule will only be applied for mobile, we don't need to support IE there.

I will take care of the other issues and if @kjellr thinks we are good on the design side then I think this should be good to go.

enejb committed:
Revert the border-radius change since it is not needed any more.

enejb committed:
Simplify: Refactor functions to just include one function that adds the 3 required filters/actions

enejb committed:
Optimization: Only add the mobile nav script if using enbale_side_menu is set.

enejb committed:
Fix position of the menu button, update support for woo nav to work the same way as regualar nav

enejb committed:
Fix rivington botton sizes to be the same.

enejb committed:
Fix Dalston site description on mobile by applying a smaller font size.

enejb committed:
Rebuild all the other themes

enejb commented:
This is how rivington is looking now. 


<img width="394" alt="Screen Shot 2020-12-16 at 2 25 05 PM" src="https://user-images.githubusercontent.com/115071/102413927-8bf4d280-3faa-11eb-92b1-1bf9831ef35f.png">
<img width="383" alt="Screen Shot 2020-12-16 at 1 21 21 PM" src="https://user-images.githubusercontent.com/115071/102408151-0bca6f00-3fa2-11eb-8f82-0fc7f6ba13d5.png">
<img width="391" alt="Screen Shot 2020-12-16 at 1 21 16 PM" src="https://user-images.githubusercontent.com/115071/102408154-0c630580-3fa2-11eb-8b01-1332a063dcb2.png">


Dalston 
<img width="439" alt="Screen Shot 2020-12-16 at 2 26 19 PM" src="https://user-images.githubusercontent.com/115071/102414045-b777bd00-3faa-11eb-8896-2250c4077b0a.png">
<img width="397" alt="Screen Shot 2020-12-16 at 2 26 14 PM" src="https://user-images.githubusercontent.com/115071/102414049-b8105380-3faa-11eb-83b4-48b4b979138c.png">
<img width="431" alt="Screen Shot 2020-12-16 at 2 26 10 PM" src="https://user-images.githubusercontent.com/115071/102414050-b8a8ea00-3faa-11eb-8c2d-a78d472eb7fb.png">

enejb committed:
Fix the admin-bar top spacing.

This was tested on the .com side to make sure that whent he admin bar is used the top spacing works as expeted.

enejb committed:
Remove the commited .idea files.

MaggieCabrera commented on dalston/style.css:
```diff
@@ -4339,7 +4339,6 @@ a {
 		display: flex;
 		align-items: flex-start;
 		justify-content: end;
-		white-space: nowrap;
```
I believe we need this to avoid the buttons from stacking on top of each other if the site title is very long

MaggieCabrera committed:
fixed woocommerce toggle with adminbar and marketing bar present

scruffian reviewed (approved):
I've tested all the themes and it looks good in all of them. Great work 👏

MaggieCabrera committed:
changed versions

MaggieCabrera commented:
Merged and deployed (D54426-code). @kjellr if you find any adjustments that need to be made to the styles do let me know and I'll get to them. @enejb and @scruffian thanks for your work
